### PR TITLE
[openssl] Fix building in Release only mode

### PR DESCRIPTION
**Pull request by AenBleidd**
Signed-off-by: Vitalii Koshura <lestat.de.lionkur@gmail.com>

### CODE REVIEW

**Comment by AenBleidd** [comment]
This is a fix for https://github.com/microsoft/vcpkg/pull/22878#issuecomment-1090808149
@strega-nil-ms, @BillyONeal, may I have your review?

File: ports/openssl/windows/portfile.cmake [hunk 1]
```diff
@@ -139,8 +139,13 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib/engines-3")
...
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
-file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/ossl-modules/legacy.dll" "${CURRENT_PACKAGES_DIR}/debug/bin/legacy.dll")
-file(RENAME "${CURRENT_PACKAGES_DIR}/lib/ossl-modules/legacy.dll" "${CURRENT_PACKAGES_DIR}/bin/legacy.dll")
+if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
+    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/ossl-modules/legacy.dll" "${CURRENT_PACKAGES_DIR}/debug/bin/legacy.dll")
+endif()
+if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
+    file(RENAME "${CURRENT_PACKAGES_DIR}/lib/ossl-modules/legacy.dll" "${CURRENT_PACKAGES_DIR}/bin/legacy.dll")
+endif()
```
Comment (Neumann-A) [inline ports/openssl/windows/portfile.cmake:147]:
```suggestion
if(NOT VCPKG_BUILD_TYPE)
    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/ossl-modules/legacy.dll" "${CURRENT_PACKAGES_DIR}/debug/bin/legacy.dll")
endif()
file(RENAME "${CURRENT_PACKAGES_DIR}/lib/ossl-modules/legacy.dll" "${CURRENT_PACKAGES_DIR}/bin/legacy.dll")
```

Comment (AenBleidd) [reply ports/openssl/windows/portfile.cmake:147]: @Neumann-A, But then it will fail if building Debug only, no?

Comment (Neumann-A) [reply ports/openssl/windows/portfile.cmake]: The new flair in vcpkg is single configuration vs multi configuration. If you want a debug only build you setup a release toolchain with debug flags. Debug only builds are impossible with the current tool.

**Review by strega-nil-ms — APPROVED** [review]

**Comment by AenBleidd** [comment]
Merged https://github.com/microsoft/vcpkg/pull/23932 as @strega-nil-ms asked

**Review by JackBoosY — APPROVED** [review]

**Comment by winterz** [comment]
anxiously waiting for this fix to land
#24003

**Comment by strega-nil-ms** [comment]
Thanks!

**Comment by winterz** [comment]
still doesn't work.  same error as reported in https://github.com/microsoft/vcpkg/issues/24003
how can I roll back to version  1.1.1n#1?  is there a way to pass OPENSSL_VERSION to `vcpkg install openssl` ?

**Comment by AenBleidd** [comment]
@winterz, I'll take a look at this issue tonight

**Pull request merged**